pkg/network/tracer/tracer.go

```diff
@@ -850,3 +851,17 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer) *usm.Monitor {
 
 	return monitor
 }
+
+// GetNetworkID retrieves the vpc_id (network_id) from IMDS
+func (t *Tracer) GetNetworkID() (string, error) {
+	id := ""
+	err := kernel.WithRootNS(kernel.ProcFSRoot(), func() error {
+		var err error
+		id, err = ec2.GetNetworkID(context.TODO())
```

Comment:
Can you pass a context into this and have the HTTP handler pass in the request context? 

Probably overkill, but that will make sure this gets correctly cancelled if the HTTP request calling this cancelled.

Reply:
sure, that makes sense